llvm/lib/Target/X86/X86FrameLowering.cpp

```diff
@@ -2958,15 +3066,30 @@ bool X86FrameLowering::restoreCalleeSavedRegisters(
   }
 
   // POP GPRs.
-  unsigned Opc = STI.is64Bit() ? X86::POP64r : X86::POP32r;
+  MachineInstrBuilder MIB;
+  bool IncompletePop2 = false;
```

Comment:
Maybe use the iterator for CST and increase one more time for Push2Pop2?

Reply:
I tried this method. It does not simplify the code b/c the odd/even check and boundary check.

Reply:
It can simplify for Push2Pop2 at least, e.g.,
```
if (X86FI->isCandidateForPush2Pop2(Reg)) {
  MIB = BuildMI(MBB, MI, DL, TII.get(getPOP2Opcode(STI)), Reg)
            .addReg(++I->getReg())
            .setMIFlag(MachineInstr::FrameDestroy);
}
```

Reply:
Done. Thanks. It does simplify the code a lot